### PR TITLE
Handle error thrown by interpolated code in pipeless script tag

### LINKED ISSUE #169: Bug: Plugin removes unescaped interpolations
## Info

| Tool      | Version                 |
| --------- | ----------------------- |
| Plugin    | v1.13.0                  |
| Prettier  | v2.2.1                  |
| Framework | none |
| Node      | v14.14.0                  |
| OS        | linux           |

## Prettier config

```json
{}
```

## Input

```pug
script.
    doSomething(!{JSON.stringify(pugLocalVariable)});
```

## Output or Error

```
Unhandled token for pipeless script tag: {"type":"interpolated-code","loc":{"start":{"line":2,"column":17},"end":{"line":2,"column":52}},"mustEscape":false,"buffer":true,"val":"JSON.stringify(pugLocalVariable)"}
```

```pug
script.
  doSomething();
```

## Expected Output

`prettier` should not raise an error or remove the interpolation.

## Additional Context

This bug is a regression; version 1.10.1 of `@prettier/plugin-pug` doesn't have it.

### PR DESCRIPTION
**Pull request by Shinigami92**
closes #169

### CODE REVIEW

**Comment by Shinigami92** [comment]
@ST-DDT I would be very happy if you can review my work in `src/printer.ts start-pipeless-text`

I'm also thinking about to create test cases to check these warnings, but:
1. it could be overkill to test it
2. it could be hard to get the stdout from console in `jest` (test-suite)

File: src/printer.ts [hunk 1]
```diff
@@ -1108,16 +1110,61 @@ export class PugPrinter {
...
 
 					index++;
 					tok = this.tokens[index];
 				}
 
-				result = format(rawText, { parser, ...this.codeInterpolationOptions });
+				try {
+					result = format(rawText, { parser, ...this.codeInterpolationOptions });
+				} catch (error: unknown) {
+					if (usedInterpolatedCode) {
```
Comment (ST-DDT) [inline src/printer.ts:1134]: I would probably invert this `if` to make it easier to distiguish when it's going to throw and when to warn.

````js
if (!usedInterpolatedCode) {
    [...]
    throw error;
}
// Continue without formatting the content
if (types.isNativeError(error)) {
[...]
````

**Review by ST-DDT — APPROVED** [review]
There are two minor things you might want to change, otherwise it's ready to be merged.

File: src/printer.ts [hunk 2]
```diff
@@ -1108,16 +1110,61 @@ export class PugPrinter {
...
+								'[PugPrinter:start-pipeless-text]:',
+								`Unexpected error for parser ${parser}.`,
+								'You used interpolated code in your pipeless script tag, so you may ignore this warning.',
+								`code: \`${rawText.trim()}\``,
+								error
+							);
+						}
+					} else {
+						logger.error(error);
+						throw error;
```
[the remark this answers is not in the document]
Comment (Shinigami92) [reply src/printer.ts:1154]: The code is already included in the error

With extra code:  
image

Without extra code:  
image

File: src/printer.ts (hunk 1 above)
Comment (Shinigami92) [reply src/printer.ts]: So I have restructured the warning context

The result looks something like this:  
image

Any further improvements?

**Review by ST-DDT — APPROVED** [review]
LGTM

**Pull request merged**